### PR TITLE
add DTO for CWMS Property

### PR DESCRIPTION
**Pull request by adamkorynta**
includes unit tests for in-memory serialization/deserialization in addition to deserialization from disk

### CODE REVIEW

File: cwms-data-api/src/main/java/cwms/cda/data/dto/Property.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            throw new FieldException("The 'category' field of a Property cannot be null or empty.");
+        }
+        if (this.name == null || this.name.trim().isEmpty()) {
+            throw new FieldException("The 'name' field of a Property cannot be null or empty.");
+        }
+        if (this.office == null || this.office.trim().isEmpty()) {
+            throw new FieldException("The 'office' field of a Property cannot be null or empty.");
+        }
+        if (this.value == null || this.value.trim().isEmpty()) {
+            throw new FieldException("The 'value' field of a Property cannot be null or empty.");
```
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/data/dto/Property.java]: This seems... odd?

Comment (rma-rripken) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/Property.java]: Which part?  That the validate does anything at all or that value can't be null/empty?

Comment (MikeNeilson) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/Property.java]: I just checked, prop_value and prop_comment are both Nullable in the database.

File: cwms-data-api/src/test/resources/cwms/cda/data/dto/property.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "category": "TestCategory",
+  "name": "TestName",
+  "office": "TestOffice",
```
Comment (rma-rripken) [inline cwms-data-api/src/test/resources/cwms/cda/data/dto/property.json]: lots of cda stuff is office-id and not office

File: cwms-data-api/src/main/java/cwms/cda/data/dto/Property.java (hunk 1 above)
Comment (MikeNeilson) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/Property.java]: We should also probably have the comment here, maybe hide it behind an option for the endpoint, general get and use value doesn't need it, but a GUI that lists them all for configuration would.

Comment (MikeNeilson) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/Property.java]: > Which part? That the validate does anything at all or that value can't be null/empty?

That value can't be null or empty. Both are a valid state for a property.

Comment (adamkorynta) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/Property.java]: good reminder that I forgot the comment. My understanding is that this method is for validating client submissions, what is the case where we'd accept a null property value? If we want to delete, we'd use the delete function.

File: cwms-data-api/src/test/resources/cwms/cda/data/dto/property.json (hunk 2 above)
Comment (adamkorynta) [reply cwms-data-api/src/test/resources/cwms/cda/data/dto/property.json]: confusing that the query parameter is office and the DTO is office-id.
What about "name"? I see a similar mismatch and inconsistency among DTO's there

File: cwms-data-api/src/main/java/cwms/cda/data/dto/Property.java (hunk 1 above)
Comment (MikeNeilson) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/Property.java]: Hmm, fair point, but "Property EXISTS and is set to NULL" does equal "Property does not exist". 
One is far more extreme than the other.

Comment (adamkorynta) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/Property.java]: ok, updated to remove validation on value

**Review by MikeNeilson — APPROVED** [review]

**Pull request merged**